### ISSUE TITLE
Keras and multiple labels

### ISSUE THREAD

**Issue by lschaupp** [keras-team/keras#10983]
Hey,

my network config is as follows:

```
def create_base_network(input_shape):
    '''Base network to be shared (eq. to feature extraction).
    '''
    input = Input(shape=input_shape)
    x = Conv2D(32, kernel_size=(3, 3), padding='same',activation='relu')(input) #(height,width)
    x = MaxPooling2D(pool_size=(1, 4), padding='same')(x)
......
    
    # Output localization vector
    v = Dense(128, activation="relu")(x) 
    x = Dense(1024, activation="relu")(v)
    x = Dense(1, activation="relu")(x)
    return Model(input, (x, v))
```

where the output of the network is x and v. I want to backpropagate two different losses: one for v and one for x. I've created a custom loss function and generated custom labels (using a generator) as following:
each label is a numpy array containing three different entries
```
label = np.array[0,0,0]
labels = np.array(labels)

```
my custom loss function looks like this, whereas I just want to use the first entry of each ground truth label and the first entry (x) of the network output:

```
def loss(y_true, y_pred):
    margin = 8
    l1 = 0.5*K.mean((1-y_true[:,0]) * K.square(y_pred[:,0]) +
                      (y_true) * K.square(K.maximum(margin - y_pred[:,0], 0)))
    return l1
```
I do not get any error (training works) but wonder if the implementation is correct...

Edit:
The shape of y_pred: (2, ?, 1)
and y_true: (?, ?)

Thanks
